src/net_processing.cpp

```diff
@@ -237,6 +248,30 @@ namespace {
 } // namespace
 
 namespace {
+/**
+ * Maintain state about nodes, protected by our own lock. Historically we put all
+ * peer tracking state in CNodeState, however this results in significant cs_main
+ * contention. Thus, new state tracking should go here, and we should eventually
+ * move most (non-validation-specific) state here.
```

Comment:
I think you should define more difference between validation-state and non-validation-state. Right now peer tracking state is spread among multiple class like `CNode` with `TxRelay` or `CNodeState` and now `CPeerState`. We should have a clear idea of _what_ should go _where_, according to which thread uses it. You should also explain how `CPeerState` aims to reduce `cs_main` contention with regards to the new threading model.